### PR TITLE
Add default option to MorphAnalysis.get

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Similar to `dict`, allow a `default` option for `MorphAnalysis.get` for the user to provide a default return value if the field is not found. The default return value remains `[]`, which is not the same as `dict.get`, but is already established as this method's default return value with the return type `List[str]`. However the new `default` option does not enforce that the user-provided default is actually `List[str]`.

### Types of change

Enhancement.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/tests/doc/test_morphanalysis.py [hunk 1]
```diff
@@ -32,7 +32,8 @@ def test_token_morph_key(i_has):
 
 def test_morph_props(i_has):
     assert i_has[0].morph.get("PronType") == ["prs"]
-    assert i_has[1].morph.get("PronType") == []
```
Comment (svlandeg) [inline spacy/tests/doc/test_morphanalysis.py:35]: Why did you remove the original assert, I thought that behaviour would still be valid?

**Review by svlandeg — COMMENTED** [review]
Looks good! Just had one small question.

File: spacy/tests/doc/test_morphanalysis.py (hunk 1 above)
Comment (adrianeboyd) [reply spacy/tests/doc/test_morphanalysis.py:35]: No reason, I'll add it back.

**Comment by bryant1410** [comment]
Thanks, folks!
